mutable_tree.go

```diff
@@ -501,3 +501,42 @@ func (tree *MutableTree) addOrphans(orphans []*Node) {
 		tree.orphans[string(node.hash)] = node.version
 	}
 }
+
+// NewSliceAt creates a copy of the current tree at given version. The same result
+// as deleting all other versions with DeleteVersion.
+// Uses input db as the backing database.
+func (tree *MutableTree) NewSliceAt(version int64, newDb dbm.DB) ([]byte, int64, *MutableTree, error) {
+	_, _, err := tree.SaveVersion()
+	if err != nil {
+		return nil, 0, nil, cmn.NewError("Saving tree: %v", err)
+	}
```

Comment:
Safe to assume the tree will have at least one version in our use case.

Reply:
This is to ensure the tree is loaded onto the backing DB and all keys have `persisted` flag set to true.

Reply:
It would be possable make this optional with an input flag as SaveVersion might be broken or take too long. But without this the final databse could end up being currupted -  might be possbale to mitigate this in outher ways.

Reply:
Wouldn't the nodes in a previously saved `ImmutableTree` (be it `MutableTree.lastSaved` or anything returned by `MutableTree.GetImmutable`) all have `persisted == true`?

Reply:
The problem is new unsaved changes. These will have persisted = false, but not be in the version we want to keep.

Reply:
Changed to just put a warning in the code that there should be no unsaved transactions?